pkg/k8shandler/elasticsearch.go

```diff
@@ -0,0 +1,328 @@
+		if transientBody["cluster"] != nil {
+			clusterBody := transientBody["cluster"].(map[string]interface{})
+			if clusterBody["routing"] != nil {
+				routingBody := clusterBody["routing"].(map[string]interface{})
+				if routingBody["allocation"] != nil {
+					allocationBody := routingBody["allocation"].(map[string]interface{})
+					if allocationBody["enable"] != nil {
+						allocation = allocationBody["enable"].(string)
+					}
+				}
+			}
+		}
+	}
+
+	return allocation, payload.Error
+}
+
+func SetMinMasterNodes(clusterName, namespace string, numberMasters int32) (bool, error) {
+
+	payload := &esCurlStruct{
+		Method:      http.MethodPut,
+		Uri:         "_cluster/settings",
+		RequestBody: fmt.Sprintf("{%q:{%q:%d}}", "persistent", "discovery.zen.minimum_master_nodes", numberMasters),
+	}
+
+	curlESService(clusterName, namespace, payload)
+
+	acknowledged := false
+	if payload.ResponseBody["acknowledged"] != nil {
+		acknowledged = payload.ResponseBody["acknowledged"].(bool)
+	}
+
+	return (payload.StatusCode == 200 && acknowledged), payload.Error
+}
+
+func GetMinMasterNodes(clusterName, namespace string) (int32, error) {
+
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		Uri:    "_cluster/settings",
+	}
+
+	curlESService(clusterName, namespace, payload)
+
+	masterCount := int32(0)
+	if payload.ResponseBody["persistent"] != nil {
+		persistentBody := payload.ResponseBody["persistent"].(map[string]interface{})
+		if persistentBody["discovery.zen.minimum_master_nodes"] != nil {
+			masterCount = int32(persistentBody["discovery.zen.minimum_master_nodes"].(float64))
+		}
+	}
+
+	return masterCount, payload.Error
+}
+
+func GetClusterHealth(clusterName, namespace string) (string, error) {
+
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		Uri:    "_cluster/health",
+	}
+
+	curlESService(clusterName, namespace, payload)
+
+	status := ""
+	if payload.ResponseBody["status"] != nil {
+		status = payload.ResponseBody["status"].(string)
+	}
+
+	return status, payload.Error
+}
+
+func GetClusterNodeCount(clusterName, namespace string) (int32, error) {
+
+	payload := &esCurlStruct{
+		Method: http.MethodGet,
+		Uri:    "_cluster/health",
+	}
+
+	curlESService(clusterName, namespace, payload)
+
+	nodeCount := int32(0)
+	if payload.ResponseBody["number_of_nodes"] != nil {
+		nodeCount = int32(payload.ResponseBody["number_of_nodes"].(float64))
+	}
+
+	return nodeCount, payload.Error
+}
+
+// TODO: also check that the number of shards in the response > 0?
+func DoSynchronizedFlush(clusterName, namespace string) (bool, error) {
+
+	payload := &esCurlStruct{
+		Method: http.MethodPost,
+		Uri:    "_flush/synced",
+	}
+
+	curlESService(clusterName, namespace, payload)
+
+	return (payload.StatusCode == 200), payload.Error
+}
+
+// This will curl the ES service and provide the certs required for doing so
+//  it will also return the http and string response
+func curlESService(clusterName, namespace string, payload *esCurlStruct) {
```

Comment:
Maybe an additional refactoring, and @josefkarasek mentioned investigation of a client, but I can see a clever fluent style pattern with something like:
`esCluster.post().withpayload(payload).do()`  or `esCluster.withpayload(payload).post()`